`<memory>`: `ranges::uninitialized_copy()` emits error C3861: '`_Rewrap_iterator`': identifier not found

```
C:\Temp>type meow.cpp
```
```cpp
#ifdef WORKAROUND
#include <algorithm>
#endif // WORKAROUND

#include <array>
#include <cassert>
#include <memory>
using namespace std;

int main() {
    const array<int, 5> src{11, 22, 33, 44, 55};
    array<int, 5> dst{-1, -1, -1, -1, -1};

    ranges::uninitialized_copy(src, dst);

    assert(dst[2] == 33);
}
```
```
C:\Temp>cl /EHsc /nologo /W4 /std:c++latest meow.cpp && meow
meow.cpp
D:\GitHub\STL\out\build\x86\out\inc\memory(68): error C3861: '_Rewrap_iterator': identifier not found
D:\GitHub\STL\out\build\x86\out\inc\memory(68): note: '_Rewrap_iterator': function was not declared in the template definition context and can be found only via argument-dependent lookup in the instantiation context
meow.cpp(14): note: see reference to function template instantiation 'std::ranges::in_out_result<std::_Array_const_iterator<_Ty,5>,std::_Array_iterator<_Ty,5>> std::ranges::_Uninitialized_copy_fn::operator ()<const std::array<_Ty,5>&,std::array<_Ty,5>&>(_Rng1,_Rng2) const' being compiled
        with
        [
            _Ty=int,
            _Rng1=const std::array<int,5> &,
            _Rng2=std::array<int,5> &
        ]

C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /DWORKAROUND meow.cpp && meow
meow.cpp

C:\Temp>
```

Definition:
https://github.com/microsoft/STL/blob/4ee3f91e06956ed2d8da4adcb8fcbf6bd75dc3b9/stl/inc/algorithm#L218-L220

Usage:
https://github.com/microsoft/STL/blob/4ee3f91e06956ed2d8da4adcb8fcbf6bd75dc3b9/stl/inc/memory#L68

`_Rewrap_iterator` definitely needs to be defined in a centralized location.

Reported as DevCom-1394583 / VSO-1310327 / AB#1310327 .

--- a/stl/inc/algorithm
+++ b/stl/inc/algorithm
@@ -215,20 +215,6 @@ namespace ranges {
         }
     }
 
-    // FUNCTION TEMPLATE _Rewrap_iterator
-    template <forward_range _Rng, class _It>
-    _NODISCARD constexpr iterator_t<_Rng> _Rewrap_iterator(_Rng&& _Range, _It&& _Val) {
-        _STL_INTERNAL_STATIC_ASSERT(is_same_v<remove_cvref_t<_It>, _Unwrapped_t<iterator_t<_Rng>>>);
-
-        if constexpr (is_same_v<remove_cvref_t<_It>, iterator_t<_Rng>>) {
-            return _STD forward<_It>(_Val);
-        } else {
-            auto _Result = _RANGES begin(_Range);
-            _Result._Seek_to(_STD forward<_It>(_Val));
-            return _Result;
-        }
-    }
-
     // FUNCTION TEMPLATE _Get_final_iterator_unwrapped
     template <forward_iterator _Wrapped, sentinel_for<_Wrapped> _Se>
     _NODISCARD constexpr auto _Get_final_iterator_unwrapped(const _Unwrapped_t<_Wrapped>& _UFirst, _Se&& _Last) {

--- a/stl/inc/xutility
+++ b/stl/inc/xutility
@@ -4766,6 +4766,20 @@ _NODISCARD bool equal(_ExPo&& _Exec, const _FwdIt1 _First1, const _FwdIt1 _Last1
 
 #ifdef __cpp_lib_concepts
 namespace ranges {
+    // FUNCTION TEMPLATE _Rewrap_iterator
+    template <forward_range _Rng, class _It>
+    _NODISCARD constexpr iterator_t<_Rng> _Rewrap_iterator(_Rng&& _Range, _It&& _Val) {
+        _STL_INTERNAL_STATIC_ASSERT(is_same_v<remove_cvref_t<_It>, _Unwrapped_t<iterator_t<_Rng>>>);
+
+        if constexpr (is_same_v<remove_cvref_t<_It>, iterator_t<_Rng>>) {
+            return _STD forward<_It>(_Val);
+        } else {
+            auto _Result = _RANGES begin(_Range);
+            _Result._Seek_to(_STD forward<_It>(_Val));
+            return _Result;
+        }
+    }
+
     // CONCEPT _Convertible_from
     template <class _To, class _From>
     concept _Convertible_from = convertible_to<_From, _To>;

--- a/tests/std/tests/P0896R4_ranges_alg_uninitialized_copy/test.cpp
+++ b/tests/std/tests/P0896R4_ranges_alg_uninitialized_copy/test.cpp
@@ -1,11 +1,22 @@
 // Copyright (c) Microsoft Corporation.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+#include <memory>
+
+// Regression test for GH-1830, in which ranges::uninitialized_copy used
+// _Rewrap_iterator which was defined in <algorithm> (now in <xutility>).
+
+void unused_function() {
+    const int src[] = {11, 22, 33, 44, 55};
+    int dst[5];
+
+    std::ranges::uninitialized_copy(src, dst);
+}
+
 #include <algorithm>
 #include <cassert>
 #include <concepts>
 #include <cstdlib>
-#include <memory>
 #include <ranges>
 #include <span>
 #include <utility>

